planner/core/find_best_task.go

```diff
@@ -2179,19 +2179,37 @@ func (ds *DataSource) convertToTableScan(prop *property.PhysicalProperty, candid
 			}
 		}
 	}
+
+	originPropTaskType := prop.TaskTp
+	if ts.StoreType == kv.TiFlash && prop.TaskTp != property.MppTaskType && ds.SCtx().GetSessionVars().IsMPPAllowed() {
+		// this prop case want to generate tiflash cop task, so change it mppTaskType to detect whether it works.
+		prop.TaskTp = property.MppTaskType
+	}
```

Comment:
explanation
CN
由于 table scan 并不是非叶节点，其自身并不会并不会枚举向下要求的 child property, 所以作为非叶节点的上层所有逻辑算子，需要枚举所有能接受的孩子的 property。所以 table scan 必定能够接受到来上自父亲节点的 mpp task type 和 TiFlash cop task 的双重枚举，有没有可能在上层 mpp task 就被减枝掉了？除了 allow_mpp = false 以外，没有任何理由，除非算子在枚举孩子 child property 时漏了，这个应该不可能。所以这里只要判断一下 allow_mpp 打开没有就知道没有其他 mpp_task 的枚举了。

Eng:
Since table scan is not a non-leaf node, it does not enumerate the child properties that are required downwards, so as all logical operators on the upper layer of non-leaf nodes, it is necessary to enumerate all acceptable child properties. Therefore, table scan must be able to accept the double enumeration of mpp task type and TiFlash cop task from the parent node. Is it possible that the mpp task emum is cut off by some factor from the upper layer ? There is no reason other than allow_mpp = false, unless the operator is missing that itself when enumerating the child property, which should not be possible. So here you only need to judge whether allow_mpp is open or not, and you will know the other enumerations of mpp_task sure be existed.